docs/MISC.md

```diff
@@ -0,0 +1,184 @@
+# Miscellaneous Topics
+
+## Enabling FP16 for Shader Use on AMD Vega
+
+### Background
+
+After my first pass at the checkerboard resolve shader, I wanted to investigate
+ways to relieve register pressure in order to squeeze in another wavefront. FP16
+came to mind since I'd be able to half my register usage in certain cases, and
+it _seemed_ like it would be relatively easy to implement. Just replace `float`
+with `half` and that's it, right?
+
+I started by searching for any articles/blogs documenting how to use FP16 with
+HLSL, DXC, SPIR-V, Vulkan, and AMD. I found two blog posts:
+
+* [mjp - Half The Precision, Twice The Fun: Working With FP16 In HLSL][mjp_fp16]
+* [AMD GPUOpen - First Steps When Implementing FP16](https://gpuopen.com/learn/first-steps-implementing-fp16/)
+
+`mjp`'s blog is far superior to the AMD blog, which elided over a lot of the
+actual implementation behaviors. The GPUOpen post doesn't even mention DXC,
+Vulkan or SPIR-V.
+
+From these blogs, I learned that it wouldn't be anywhere as simple as I hoped,
+mostly because I learned that `half` maps right to `float` in HLSL. My co-worker
+[aellem](https://github.com/aellem) pointed me to an
+[HLSL doc page on scalar types](https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-scalar).
+
+For me to use FP16 in my shader, it seemed I'd have to do two things:
+
+1. Generate FP16 code from my HLSL
+   1. Use `min16float*` types to request minimum (aka relaxed or flexible)
+      16-bit precision for floats. The underlying implementation can 'upgrade'
+      them to 32-bit precision since I'm only requesting a lower bound on
+      precision.
+   1. Force 16-bit precision on `half` by using `-enable-16bit-types` when
+      compiling my shaders with DXC.
+1. Enable FP16 extensions on my Vulkan device
```

Comment:
nit: odd numbering scheme?

Reply:
It's a 'best practice': https://guides.github.com/features/mastering-markdown/

Markdown doesn't actually care _what specific_ number you use, it only cares that you use _a_ number. It renders the ordered/numbered list independent of the input numerals. I think the idea is that if you have to go back and edit the list, you can just add an extra item w/o worrying about fixing the other numbers. But then it's weird to read if you didn't know about it